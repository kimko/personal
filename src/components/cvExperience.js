import React from "react";
import PropTypes from "prop-types";

export const Task = (props) => {
  const classn = `f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black`;
  return (
    <li className="pb2 mw6">
      <div>
        <span className="mr1">{props.task}</span>
        {props.tools.map((item, idx) => (
          <span key={idx} className={`${classn}`}>
            {item}
          </span>
        ))}
      </div>
    </li>
  );
};

Task.propTypes = {
  task: PropTypes.string,
  tools: PropTypes.array,
};

export const Job = (props) => {
  return (
    <article className="pt2 f6">
      <div className="flex flex-wrap">
        <div className="w-100 w-third-ns">
          <h1 className="cf f6 mv1">
            <time className="fl" dateTime="2018-07">
              {props.job.start}
            </time>
            <span className="fl ph1">-</span>
            <time className="fl" dateTime="2021-03">
              {props.job.end}
            </time>
          </h1>
          <div className="fl f6">{props.job.location}</div>
        </div>
        <div className="w-100 w-two-thirds-ns pl4-ns">
          <h1 className=" f6 mv1 pr2">
            <span className="">{props.job.title}</span>
            <span className="f7 ph1">-</span>
            <span className="f7">{props.job.company}</span>
          </h1>
          <ul className="pl3 pr2 mt0">
            {props.job.experience.map((item, idx) => (
              <Task key={idx} task={item.task} tools={item.tools} />
            ))}
          </ul>
        </div>
      </div>
    </article>
  );
};

Job.propTypes = {
  job: PropTypes.object,
};

export const Experience = (props) => {
  return (
    <section
      className="bb w-100 cf pl4 pb2"
      style={props.style}
      data-testid="test-section-experience"
    >
      <div className="flex flex-wrap">
        <div className="w-100 w-third-ns">
          <h1 className="bl bw2 f5 pl1">Experience</h1>
        </div>
      </div>
      {props.jobData.map((job, idx) => (
        <Job key={idx} style={props.style} job={job} />
      ))}
    </section>
  );
};

Experience.propTypes = {
  style: PropTypes.object,
  jobData: PropTypes.array,
};
