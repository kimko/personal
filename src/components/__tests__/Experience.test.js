import React from "react";
import { render, cleanup } from "@testing-library/react";

import { Experience } from "../cvExperience";

afterEach(cleanup);

const jobData = [
  {
    start: "2020",
    end: "Present",
    title: "Cat Herder",
    company: "Cats'r'Us",
    location: "Portland, OR",
    experience: [
      {
        task: `Play with Cats.`,
        tools: ["yarn", "flashlight"],
      },
      {
        task: `Fed the Cats`,
        tools: ["Milk", "Kibble"],
      },
    ],
  },
  {
    start: "2018",
    end: "2020",
    title: "Hahum",
    company: "Hmm",
    location: "Portland, OR",
    experience: [
      {
        task: `Task task task task task task.`,
        tools: ["tool1", "tool2", "tool3"],
      },
      {
        task: `asdf jkl;`,
        tools: ["a", "b", "d"],
      },
    ],
  },
];

it("renders a Experience section", () => {
  const { getByTestId, asFragment } = render(<Experience jobData={jobData} />);

  expect(getByTestId("test-section-experience")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
