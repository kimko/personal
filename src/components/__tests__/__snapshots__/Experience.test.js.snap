// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Experience section 1`] = `
<DocumentFragment>
  <section
    class="bb w-100 cf pl4 pb2"
    data-testid="test-section-experience"
  >
    <div
      class="flex flex-wrap"
    >
      <div
        class="w-100 w-third-ns"
      >
        <h1
          class="bl bw2 f5 pl1"
        >
          Experience
        </h1>
      </div>
    </div>
    <article
      class="pt2 f6"
    >
      <div
        class="flex flex-wrap"
      >
        <div
          class="w-100 w-third-ns"
        >
          <h1
            class="cf f6 mv1"
          >
            <time
              class="fl"
              datetime="2018-07"
            >
              2020
            </time>
            <span
              class="fl ph1"
            >
              -
            </span>
            <time
              class="fl"
              datetime="2021-03"
            >
              Present
            </time>
          </h1>
          <div
            class="fl f6"
          >
            Portland, OR
          </div>
        </div>
        <div
          class="w-100 w-two-thirds-ns pl4-ns"
        >
          <h1
            class=" f6 mv1 pr2"
          >
            <span
              class=""
            >
              Cat Herder
            </span>
            <span
              class="f7 ph1"
            >
              -
            </span>
            <span
              class="f7"
            >
              Cats'r'Us
            </span>
          </h1>
          <ul
            class="pl3 pr2 mt0"
          >
            <li
              class="pb2 mw6"
            >
              <div>
                <span
                  class="mr1"
                >
                  Play with Cats.
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  yarn
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  flashlight
                </span>
              </div>
            </li>
            <li
              class="pb2 mw6"
            >
              <div>
                <span
                  class="mr1"
                >
                  Fed the Cats
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  Milk
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  Kibble
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </article>
    <article
      class="pt2 f6"
    >
      <div
        class="flex flex-wrap"
      >
        <div
          class="w-100 w-third-ns"
        >
          <h1
            class="cf f6 mv1"
          >
            <time
              class="fl"
              datetime="2018-07"
            >
              2018
            </time>
            <span
              class="fl ph1"
            >
              -
            </span>
            <time
              class="fl"
              datetime="2021-03"
            >
              2020
            </time>
          </h1>
          <div
            class="fl f6"
          >
            Portland, OR
          </div>
        </div>
        <div
          class="w-100 w-two-thirds-ns pl4-ns"
        >
          <h1
            class=" f6 mv1 pr2"
          >
            <span
              class=""
            >
              Hahum
            </span>
            <span
              class="f7 ph1"
            >
              -
            </span>
            <span
              class="f7"
            >
              Hmm
            </span>
          </h1>
          <ul
            class="pl3 pr2 mt0"
          >
            <li
              class="pb2 mw6"
            >
              <div>
                <span
                  class="mr1"
                >
                  Task task task task task task.
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  tool1
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  tool2
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  tool3
                </span>
              </div>
            </li>
            <li
              class="pb2 mw6"
            >
              <div>
                <span
                  class="mr1"
                >
                  asdf jkl;
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  a
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  b
                </span>
                <span
                  class="f7 near-black ttl pr1 dib ba b--dotted mr1 tracked-tight
        bg-animate hover-bg-gray hover-near-black"
                >
                  d
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </article>
  </section>
</DocumentFragment>
`;
